pkg/playback/replayer.go

```diff
@@ -160,9 +159,8 @@ func (replayer *Replayer) handler(w http.ResponseWriter, r *http.Request) {
 	go func() {
 		// Note: if there are any errors in processing recorded webhooks here,
 		// we log the error and keep going.
-		replayer.replayLock.Wait()       // only send webhooks after the previous request/response is handled
-		replayer.replayLock.Add(1)       // acquire lock so we can send webhooks
-		defer replayer.replayLock.Done() // release lock when done sending webhooks
+		replayer.replayLock.Lock()         // only send webhooks after the previous request/response is handled
+		defer replayer.replayLock.Unlock() // release lock when done sending webhooks
```

Comment:
This LGTM, but I think we might also want to explore (as a followup) a more explicit (non-mutex based) way to implement what's essentially a pipeline (receive request -> send response -> send webhooks -> wait for next request).

Reply:
yup I agree! I began some of that exploration & I'll spend some more time on it!